File: api/langs/java/plump_message/Card.java
package plump_message;


public final class Card {
    public final @com.novi.serde.Unsigned Long suit;
    public final @com.novi.serde.Unsigned Long value;

    public Card(@com.novi.serde.Unsigned Long suit, @com.novi.serde.Unsigned Long value) {
        java.util.Objects.requireNonNull(suit, "suit must not be null");
        java.util.Objects.requireNonNull(value, "value must not be null");
        this.suit = suit;
        this.value = value;
    }

    public void serialize(com.novi.serde.Serializer serializer) throws com.novi.serde.SerializationError {
        serializer.increase_container_depth();
        serializer.serialize_u64(suit);
        serializer.serialize_u64(value);
        serializer.decrease_container_depth();
    }

    public byte[] bincodeSerialize() throws com.novi.serde.SerializationError {
        com.novi.serde.Serializer serializer = new com.novi.bincode.BincodeSerializer();
        serialize(serializer);
        return serializer.get_bytes();
    }

    public static Card deserialize(com.novi.serde.Deserializer deserializer) throws com.novi.serde.DeserializationError {
        deserializer.increase_container_depth();
        Builder builder = new Builder();
        builder.suit = deserializer.deserialize_u64();
        builder.value = deserializer.deserialize_u64();
        deserializer.decrease_container_depth();
        return builder.build();
    }

    public static Card bincodeDeserialize(byte[] input) throws com.novi.serde.DeserializationError {
        if (input == null) {
             throw new com.novi.serde.DeserializationError("Cannot deserialize null array");
        }
        com.novi.serde.Deserializer deserializer = new com.novi.bincode.BincodeDeserializer(input);
        Card value = deserialize(deserializer);
        if (deserializer.get_buffer_offset() < input.length) {
             throw new com.novi.serde.DeserializationError("Some input bytes were not read");
        }
        return value;
    }

    public boolean equals(Object obj) {
        if (this == obj) return true;
        if (obj == null) return false;
        if (getClass() != obj.getClass()) return false;
        Card other = (Card) obj;
        if (!java.util.Objects.equals(this.suit, other.suit)) { return false; }
        if (!java.util.Objects.equals(this.value, other.value)) { return false; }
        return true;
    }

    public int hashCode() {
        int value = 7;
        value = 31 * value + (this.suit != null ? this.suit.hashCode() : 0);
        value = 31 * value + (this.value != null ? this.value.hashCode() : 0);
        return value;
    }

    public static final class Builder {
        public @com.novi.serde.Unsigned Long suit;
        public @com.novi.serde.Unsigned Long value;

        public Card build() {
            return new Card(
                suit,
                value
            );
        }
    }
}

File: api/langs/csharp/plump_message/Card.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Numerics;

namespace plump_message {

    public sealed class Card: IEquatable<Card>, ICloneable {
        public ulong suit;
        public ulong value;

        public Card(ulong _suit, ulong _value) {
            suit = _suit;
            value = _value;
        }

        public void Serialize(Serde.ISerializer serializer) {
            serializer.increase_container_depth();
            serializer.serialize_u64(suit);
            serializer.serialize_u64(value);
            serializer.decrease_container_depth();
        }

        public int BincodeSerialize(byte[] outputBuffer) => BincodeSerialize(new ArraySegment<byte>(outputBuffer));

        public int BincodeSerialize(ArraySegment<byte> outputBuffer) {
            Serde.ISerializer serializer = new Bincode.BincodeSerializer(outputBuffer);
            Serialize(serializer);
            return serializer.get_buffer_offset();
        }

        public byte[] BincodeSerialize()  {
            Serde.ISerializer serializer = new Bincode.BincodeSerializer();
            Serialize(serializer);
            return serializer.get_bytes();
        }

        public static Card Deserialize(Serde.IDeserializer deserializer) {
            deserializer.increase_container_depth();
            Card obj = new Card(
            	deserializer.deserialize_u64(),
            	deserializer.deserialize_u64());
            deserializer.decrease_container_depth();
            return obj;
        }

        public static Card BincodeDeserialize(byte[] input) => BincodeDeserialize(new ArraySegment<byte>(input));

        public static Card BincodeDeserialize(ArraySegment<byte> input) {
            if (input == null) {
                 throw new Serde.DeserializationException("Cannot deserialize null array");
            }
            Serde.IDeserializer deserializer = new Bincode.BincodeDeserializer(input);
            Card value = Deserialize(deserializer);
            if (deserializer.get_buffer_offset() < input.Count) {
                 throw new Serde.DeserializationException("Some input bytes were not read");
            }
            return value;
        }
        public override bool Equals(object obj) => obj is Card other && Equals(other);

        public static bool operator ==(Card left, Card right) => Equals(left, right);

        public static bool operator !=(Card left, Card right) => !Equals(left, right);

        public bool Equals(Card other) {
            if (other == null) return false;
            if (ReferenceEquals(this, other)) return true;
            if (!suit.Equals(other.suit)) return false;
            if (!value.Equals(other.value)) return false;
            return true;
        }

        public override int GetHashCode() {
            unchecked {
                int value = 7;
                value = 31 * value + suit.GetHashCode();
                value = 31 * value + value.GetHashCode();
                return value;
            }
        }

        /// <summary>Creates a shallow clone of the object.</summary>
        public Card Clone() => (Card)MemberwiseClone();

        object ICloneable.Clone() => Clone();

    }

} // end of namespace plump_message
